File: Swap Nodes in Pairs.js
var swapPairs = function(head) {
    if (head == null || head.next == null )
     return head;
    var p1 = new ListNode(0);
    var p2 = new ListNode(0);
    var result = new ListNode(0);
    var temp = new ListNode(0);
    
    p1 = head;
    p2 = head.next;
    result = p2;
    temp = null;
    
    while (p1 != null && p2 != null )
    {
        p1.next = p2.next;
        p2.next = p1;
        
        if (temp != null )
            temp.next = p2;
        temp = p1;
        if (p1.next != null && p1.next.next != null )
            {
                p1 = p1.next;
                p2 = p1.next;
            }
            else 
                break ;
    }
    return result;
};

File: Swap Nodes in Pairs.c
struct ListNode* swapPairs(struct ListNode* head) {
    
    if (head == NULL || head -> next == NULL )
     return head;
    
    struct ListNode *p1 = head;
    struct ListNode *p2 = head -> next;
    struct ListNode *result = p2;
    struct ListNode *temp = NULL ;
    while (p1 != NULL && p2 != NULL )
    {
        p1 -> next = p2 -> next;
        p2 -> next = p1;
        
        if (temp != NULL )
                temp -> next = p2;
        temp = p1;
        if (p1 -> next != NULL && p1 -> next -> next != NULL )
            {
                p1 = p1 -> next;
                p2 = p1 -> next;
            }
            else 
                break ;
    }
    return result;
}
